packages/sdk-cli/src/commands/connect.test.ts

```diff
@@ -56,22 +58,29 @@ let mockWS: events.EventEmitter;
 
 let hostConnections: HostConnections;
 let relayHostsSpy: jest.SpyInstance;
+let relayConnectSpy: jest.SpyInstance;
 let hostConnectSpy: jest.SpyInstance;
 
 const mockRelayHostsResponse = {
-  device: (hosts: developerRelay.Host[]) =>
+  device: (hosts: Host[]) =>
     relayHostsSpy.mockResolvedValueOnce({ appHost: hosts, companionHost: [] }),
-  phone: (hosts: developerRelay.Host[]) =>
+  phone: (hosts: Host[]) =>
     relayHostsSpy.mockResolvedValueOnce({ appHost: [], companionHost: hosts }),
 };
 
 beforeEach(() => {
   hostConnections = new HostConnections();
+
   ({ cli, mockLog, mockPrompt } = commandTestHarness(
     connect({ hostConnections }),
   ));
-  relayHostsSpy = jest.spyOn(developerRelay, 'hosts');
+  relayHostsSpy = jest.spyOn(DeveloperRelay.prototype, 'hosts');
   hostConnectSpy = jest.spyOn(hostConnections, 'connect');
+  relayConnectSpy = jest.spyOn(DeveloperRelay.prototype, 'connect');
+
+  // ASK: Works in beforeEach(), but doesn't in beforeAll()
```

Comment:
Yes, thanks for noticing! If I define relayConnectSpy in `beforeAll()` instead of `beforeEach()`, the mock just...doesn't work, the origin function is called, as if there has never been a mock defined.

```
beforeAll(() => {
  relayConnectSpy = jest.spyOn(DeveloperRelay.prototype, 'connect');
  relayConnectSpy.mockResolvedValue(new Duplex());
});
```

Reply:
Can you point me to a test that fails in this scenario so I can help you debug it?

Reply:
Almost every single one :) Because the mock simply doesn't work, every time `doConnect` is called (and it's called before/in every test), the `developerRelay.connect(host.id)` in `connectAction` sends an actual request to `https://api.fitbit.com:443/1/user/-/developer-relay/hosts/<hostType>`, which obv gives a 404

Reply:
If you need, you can choose any except "logs an error if no ..." and "does not auto-connect a busy host"

Reply:
`jest.config.js`: `restoreMocks: true` caused this.

Reply:
Noticed you'd resolved this comment, but the `ASK` comment is still present. Remove it and then resolve this again.